web/angular-wallet/src/app/main/transactions/transaction-table/transaction-table.component.ts

```diff
@@ -42,23 +50,35 @@ export class TransactionTableComponent extends UnsubscribeOnDestroy implements O
   private account: Account;
 
   @Input() dataSource: MatTableDataSource<Transaction>;
+  @Input() relatedAccount: Account;
   @Input() public displayedColumns = ['transaction_id', 'attachment', 'timestamp', 'type', 'amount', 'fee', 'account', 'confirmations'];
   @Input() paginationEnabled = true;
   @ViewChild(MatPaginator) paginator: MatPaginator;
 
   public isMultiOutPayment(transaction: Transaction): boolean {
-    return isMultiOutSameTransaction(transaction) || isMultiOutTransaction(transaction)
+    return isMultiOutSameTransaction(transaction) || isMultiOutTransaction(transaction);
   }
 
   public ngOnInit(): void {
-    this.setAccount(this.route.snapshot.data.account);
+
+    const referencedAccount = this.route.snapshot.data.account;
+    if (referencedAccount) {
+      this.setAccount(referencedAccount);
+      return;
+    }
+
     this.accountService.currentAccount
-    .pipe(
-      takeUntil(this.unsubscribeAll)
-    )
-    .subscribe((account) => {
-      this.setAccount(account);
-    });
+      .pipe(
+        takeUntil(this.unsubscribeAll)
+      )
+      .subscribe((account) => {
+        this.setAccount(account);
+      });
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
this hook updates if you click another account within the tx table of account details